presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoPreparedStatement.java

```diff
@@ -849,4 +859,59 @@ private static String typedNull(String prestoType)
     {
         return format("CAST(NULL AS %s)", prestoType);
     }
+
+    private static List<ColumnInfo> getDescribeOutputColumnInfoList(ResultSet resultSet)
+            throws SQLException
+    {
+        ImmutableList.Builder<ColumnInfo> list = ImmutableList.builder();
+        while (resultSet.next()) {
+            String columnName = resultSet.getString("Column Name");
+            String catalog = resultSet.getString("Catalog");
+            String schema = resultSet.getString("Schema");
+            String table = resultSet.getString("Table");
+            ClientTypeSignature clientTypeSignature = getClientTypeSignatureFromTypeString(resultSet.getString("Type"));
+            ColumnInfo.Builder builder = new ColumnInfo.Builder()
+                    .setColumnName(columnName)
+                    .setColumnLabel(columnName)
+                    .setCatalogName(catalog)
+                    .setSchemaName(schema)
+                    .setTableName(table)
+                    .setColumnTypeSignature(clientTypeSignature)
+                    .setNullable(ColumnInfo.Nullable.UNKNOWN);
+            setTypeInfo(builder, clientTypeSignature);
+            list.add(builder.build());
+        }
+        return list.build();
+    }
+
+    private static ClientTypeSignature getClientTypeSignatureFromTypeString(String type)
+    {
+        String topLevelTypeRegexPattern = "^(.+?)\\((.+)\\)$";
+        Pattern pattern = Pattern.compile(topLevelTypeRegexPattern);
+        Matcher matcher = pattern.matcher(type);
+        String topLevelType;
+        List<ClientTypeSignatureParameter> arguments = new ArrayList<>();
+        if (matcher.find()) {
+            topLevelType = matcher.group(1);
+            String typeParameters = matcher.group(2);
+            if (topLevelType.equals("decimal")) {
```

Comment:
Some types such as `decimal`, `char` and `varchar` need to be treated specially since `ColumnInfo#ColumnInfo` will visit the `arguments` parameter list.